catalog/spatial/wfs/spatial-wfs-metacardtype-registry/src/main/java/org/codice/ddf/spatial/ogc/wfs/catalog/metacardtype/registry/WfsMetacardTypeRegistryImpl.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.ogc.wfs.catalog.metacardtype.registry;
+
+import com.google.common.base.Verify;
+import ddf.catalog.data.MetacardType;
+import java.util.ArrayList;
+import java.util.Dictionary;
+import java.util.List;
+import java.util.Optional;
+import org.codice.ddf.configuration.DictionaryMap;
+import org.codice.ddf.spatial.ogc.wfs.catalog.metacardtype.registry.api.WfsMetacardTypeRegistry;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+
+public final class WfsMetacardTypeRegistryImpl implements WfsMetacardTypeRegistry {
+
+  public static final String FEATURE_NAME = "feature-name";
+
+  public static final String SOURCE_ID = "source-id";
+
+  private BundleContext bundleContext;
+
+  private List<ServiceRegistration<MetacardType>> serviceRegistrations;
+
+  public WfsMetacardTypeRegistryImpl(BundleContext bundleContext) {
+    this.bundleContext = bundleContext;
+    this.serviceRegistrations = new ArrayList<>();
+  }
+
+  /** {@inheritDoc} */
+  public Optional<MetacardType> lookupMetacardTypeBySimpleName(String sourceId, String simpleName) {
+
+    if (sourceId == null || simpleName == null) {
+      return Optional.empty();
+    }
+
+    Optional<ServiceRegistration<MetacardType>> serviceRegistrationOptional =
+        serviceRegistrations
+            .stream()
+            .filter(s -> s.getReference().getProperty(SOURCE_ID) != null)
+            .filter(s -> s.getReference().getProperty(SOURCE_ID).equals(sourceId))
+            .filter(s -> s.getReference().getProperty(FEATURE_NAME) != null)
+            .filter(s -> s.getReference().getProperty(FEATURE_NAME).equals(simpleName))
```

Comment:
Could we combine the != null and equals check using StringUtils.equals?
`s -> StringUtils.equals(s.getReference().getProperty(SOURCE_ID), sourceId)`